e3sm_diags/driver/utils/io.py

```diff
@@ -111,6 +111,61 @@ def _write_vars_to_netcdf(
     -----
     Replaces `e3sm_diags.driver.utils.general.save_ncfiles()`.
     """
+    _write_to_netcdf(parameter, ds_test[var_key], var_key, "test")
+
+    if ds_ref is not None:
+        _write_to_netcdf(parameter, ds_ref[var_key], var_key, "ref")
+
+    if ds_diff is not None:
+        _write_to_netcdf(parameter, ds_diff[var_key], var_key, "diff")
+
+
+def _write_to_netcdf(
+    parameter,
+    var: xr.DataArray,
+    var_key: str,
+    data_type: Literal["test", "ref", "diff"],
+):
+    dir_path = _get_output_dir(parameter)
+    filename = f"{parameter.output_file}_{data_type}.nc"
+
+    filepath = os.path.join(dir_path, filename)
+
+    var.to_netcdf(filepath)
+
+    logger.info(f"'{var_key}' {data_type} variable output saved in: {filepath}")
+
+    return filename
+
+
+def _write_vars_to_single_netcdf(
```

Comment:
This function is kept just in case we want to implement it in the future. It is cleaner to have a single netCDF with the test, ref, and diff variables.